models/staging/graph/stg_all_graph_resources.sql

```diff
@@ -10,61 +10,90 @@ with unioned as (
+        end as prefix,
+        {{ dbt_utils.concat(["'/'",'unioned.file_path']) }} as file_path_prefixed_with_slash
 
     from unioned
-    where coalesce(is_enabled, True) = True
-    and not(resource_type = 'model' and package_name = 'dbt_project_evaluator')
+), 
+
+joined as (
 
+    select
+        unioned_with_prefix.unique_id as resource_id, 
+        unioned_with_prefix.resource_name, 
+        unioned_with_prefix.prefix, 
+        unioned_with_prefix.resource_type, 
+        unioned_with_prefix.file_path, 
+        naming_convention_prefixes.model_type as model_type_prefix,
+        naming_convention_folders.model_type as model_type_folder,
+        {{ dbt_utils.position('naming_convention_folders.model_type','unioned_with_prefix.file_path_prefixed_with_slash') }} as position_folder,  
+        unioned_with_prefix.is_enabled, 
+        unioned_with_prefix.materialized, 
+        unioned_with_prefix.on_schema_change, 
+        unioned_with_prefix.database, 
+        unioned_with_prefix.schema, 
+        unioned_with_prefix.package_name, 
+        unioned_with_prefix.alias, 
+        unioned_with_prefix.is_described, 
+        unioned_with_prefix.exposure_type, 
+        unioned_with_prefix.maturity, 
+        unioned_with_prefix.url, 
+        unioned_with_prefix.metric_type, 
+        unioned_with_prefix.model, 
+        unioned_with_prefix.label, 
+        unioned_with_prefix.sql, 
+        unioned_with_prefix.timestamp as timestamp,  
+        unioned_with_prefix.source_name, -- NULL for non-source resources
+        unioned_with_prefix.is_source_described, 
+        unioned_with_prefix.loaded_at_field, 
+        unioned_with_prefix.loader, 
+        unioned_with_prefix.identifier
+
+    from unioned_with_prefix
+    left join naming_convention_prefixes
+        on unioned_with_prefix.prefix = naming_convention_prefixes.prefix_value
+
+    left join naming_convention_folders
+        -- we added '/' at the front of the path, to be /folder1/folder2/folder3/mmodel_name.sql
+        -- and we search for /naming_convention_folder/ ; this allows us to only match full folder names
+        on file_path_prefixed_with_slash like {{ dbt_utils.concat(["'%/'",'naming_convention_folders.folder_name_value',"'/%'"]) }}
+    
+    where coalesce(unioned_with_prefix.is_enabled, True) = True
+    and not(unioned_with_prefix.resource_type = 'model' and unioned_with_prefix.package_name = 'dbt_project_evaluator')
+
+), 
+
+final as (
+    select 
+        *, 
+        case 
+            when resource_type in ('test', 'source', 'metric', 'exposure', 'seed') then null
+            -- by default we will define the model type based on its prefix in the case prefix and folder types are different
+            else coalesce(model_type_prefix, model_type_folder, 'other') 
+        end as model_type,
+        row_number() over (partition by resource_id order by position_folder desc) as position_rk
```

Comment:
What does `rk` stand for?

Reply:
`rk` is short for `rank`. I just want one row in case my joins are 1-n instead of 1-1. I am filtering it in my last query (can't use qualify in all data warehouses)


Do you want me to rename the column?

Reply:
Got it! Thanks for the explanation!